src/Typecheck.elm

```diff
@@ -107,31 +108,75 @@ checkSig sig args =
       Nothing -> Invalid
 
 
+substitute : Term -> Term -> String -> Term
```

Comment:
Eventually we should look at moving towards environment-based function evaluation, however for now this will work nicely.